src/swaps/hop/HopSwapProvider.ts

```diff
@@ -0,0 +1,579 @@
+import BN from 'bignumber.js'
+import { Hop } from '@hop-protocol/sdk'
+import { chains, currencyToUnit, unitToCurrency } from '@liquality/cryptoassets'
+import { v4 as uuidv4 } from 'uuid'
+import { createClient } from 'urql'
+import { Wallet } from 'ethers'
+import { SwapProvider } from '../SwapProvider'
+import cryptoassets from '../../utils/cryptoassets'
+import { prettyBalance } from '../../utils/coinFormatter'
+import { withInterval, withLock } from '../../store/actions/performNextAction/utils'
+import { isERC20 } from '../../utils/asset'
+import {
+  BaseSwapProviderConfig,
+  EstimateFeeRequest,
+  EstimateFeeResponse,
+  NextSwapActionRequest,
+  QuoteRequest,
+  SwapRequest,
+  SwapStatus,
+} from '../types';
+import { ActionContext } from '../../store'
+import { Network} from '../../store/types'
+
+export interface HopSwapProviderConfig extends BaseSwapProviderConfig {
+  graphqlBaseURL: string;
+}
+
+class HopSwapProvider extends SwapProvider {
+  config: HopSwapProviderConfig;
+  graphqlURLs: { [key: string]: string };
+
+  constructor(config: HopSwapProviderConfig) {
+    super(config)
+    this.graphqlURLs = {
+      url: this.config.graphqlBaseURL,
+      ethereum: 'hop-mainnet',
+      xdai: 'hop-xdai',
+      arbitrum: 'hop-arbitrum',
+      polygon: 'hop-polygon',
+      optimism: 'hop-optimism'
+    }
+  }
+  /**
+   * Get the supported pairs of this provider for this network
+   * @param {{ network }} network
+   */
+  // eslint-disable-next-line no-unused-vars
+  async getSupportedPairs() {
+    return [];
+  }
+
+  confirmationsSendCount(networkName: string) {
+      const networkToConfirmations: { [key: string]: number } =    
+      {
+        ethereum: 1,
```

Comment:
How are these determined? If it's a hop parameter, they should provide this via a library?
If it's liquality defined then this is a chain configuration that can be retrieved from `cryptoassets`. It's `chains[chain].safeConfirmations`

Reply:
chains[chain].safeConfirmations will do the job. I will remove the other function